### DISCUSSION TITLE
IMage processing with native platfrom. Bidirectional conversion between Android Bitmap and iOS UIImage

### DISCUSSION THREAD

**Discussion by GaurDev06** [korlibs/korge#789] [Q&A] [answered]
Hi @soywiz,
How to convert android bitmap to korim bitmap?
And in IOS uiimage to bitmap and vice-versa?
TIA

**Comment by soywiz** [comment, marked as answer]
## Android

* https://github.com/korlibs/korge/blob/488cf0c5af58d10bcfeab9e1c17432f516a568f8/korim/src/androidMain/kotlin/com/soywiz/korim/format/NativeImageFormatProviderAndroid.kt#L129
* https://github.com/korlibs/korge/blob/488cf0c5af58d10bcfeab9e1c17432f516a568f8/korim/src/androidMain/kotlin/com/soywiz/korim/format/NativeImageFormatProviderAndroid.kt#L143

```kotlin
korimBitmap.toAndroidBitmap() // KorIM -> Android
AndroidNativeImage(androidBitmap).toBMP32() // Android -> KorIM (toBMP32 might not be required if you not need raw pixels)
```

## iOS

* `val uiImage: UIImage = bitmap32.toUIImage()`
* `val bitmap32: Bitmap32 = uiImage.toBitmap32()`

NOTE: If those methods are not available in your KorIM version, you can grab them from here and put them in your code: https://github.com/korlibs/korge/pull/1108

**Comment by Burtan** [comment]
Is any of the above iOS classes meant to be used with swift/objective-c?
My use case is kotlin multiplatform mobile and I want to draw images in SwiftUI.
The android/native conversion is straightforward, but I'm struggling with the iOS one :D.

**Reply by Burtan** [reply]
Considering this answer: https://github.com/korlibs/korge/discussions/483
I guess I have to use transferBitmap32CGContext in iOS specific kotlin code to get CGContext/CGImage in iOS swift code.

**Reply by soywiz** [reply]
Fixed here https://github.com/korlibs/korge/pull/1108/files

**Reply by Burtan** [reply]
Thanks for the fast response and fix! I'll check it out when the new version is released :-)
